test/helpers/fixtureUtils.js

```diff
@@ -0,0 +1,71 @@
+import child from "child_process";
+import os from "os";
+import path from "path";
+
+import fs from "graceful-fs";
+import { padStart } from "lodash";
+import mkdirp from "mkdirp";
+import pify from "pify";
+import rimraf from "rimraf";
+
+const CURRENT_SHA = child.execSync("git rev-parse --short HEAD", {
+  cwd: __dirname,
+  encoding: "utf8",
+}).trim();
+
+const execAsync = pify(child.exec);
+const realpathAsync = pify(fs.realpath);
+
+const _mkdirpAsync = pify(mkdirp);
+const _rimrafAsync = pify(rimraf);
+
+// graceful-fs overrides for rimraf
+const { unlink, chmod, stat, lstat, rmdir, readdir } = fs;
+
+export const mkdirpAsync = (fp) => _mkdirpAsync(fp, { fs });
+export const rimrafAsync = (fp) => _rimrafAsync(fp, { unlink, chmod, stat, lstat, rmdir, readdir });
+
+export function getTempDir(suffix) {
+  // e.g., "lerna-12345-deadbeef"
+  const prefix = [
+    "lerna",
+    process.pid,
+    CURRENT_SHA,
+  ];
+
+  if (suffix) {
+    // e.g., "lerna-12345-deadbeef-external"
+    prefix.push(suffix);
+  }
+
+  const tmpDir = path.join(os.tmpdir() || "/tmp", prefix.join("-"));
```

Comment:
Using a real temp directory helps avoid lots of subtle bugs in the tests due to "fall through" into the parent git repo. Also, it's just cleaner, and in some cases even faster (since tempdirs are often located on ramdisks).